src/test/java/com/solace/spring/cloud/stream/binder/inbound/queue/FlowXMLMessageListenerTest.java

```diff
@@ -152,4 +158,53 @@ void testStartReceiverThreads_WatchdogLogsWarningForLongProcessing() throws NoSu
         await().atMost(5500, TimeUnit.MILLISECONDS)
                 .until(() -> activeMessages.iterator().next().isErrored());
     }
+
+    @Test
+    void testStartReceiverThreads_WatchdogLogsWarningForCongestedQueue(CapturedOutput capturedOutput) {
+        FlowXMLMessageListener listener = new FlowXMLMessageListener();
+        MockClock clock = new MockClock();
```

Comment:
Don't mock, do it with the real code. maybe change the config to hit the congestion earlier